docs/ref/templates/builtins.txt

```diff
@@ -1838,18 +1838,21 @@ To escape each element of a sequence, use the :tfilter:`escapeseq` filter.
 ``escapejs``
 ------------
 
-Escapes characters for use in JavaScript strings. This does *not* make the
-string safe for use in HTML or JavaScript template literals, but does protect
-you from syntax errors when using templates to generate JavaScript/JSON.
+Escapes characters for use in JavaScript strings (text written inside single
+or double quotes). Using this filter only makes the string safe when used as
+the whole content of a JavaScript or JSON string literal (please see the
+example below). Note that it does not make the string safe for use in
+"JavaScript template literals" (i.e. the JavaScript backtick syntax), and
+any other use is not supported. It is generally recommended however that
+data should be passed using HTML ``data-`` attributes, or the
+:tfilter:`json_script` filter, rather than in embedded JavaScript.
 
 For example:
 
 .. code-block:: html+django
 
-    {{ value|escapejs }}
-
-If ``value`` is ``"testing\r\njavascript 'string\" <b>escaping</b>"``,
-the output will be ``"testing\\u000D\\u000Ajavascript \\u0027string\\u0022 \\u003Cb\\u003Eescaping\\u003C/b\\u003E"``.
+    <script>
+    let myValue = '{{ value|escapejs }}'
```

Comment:
We can indicate the code would continue, and use a more realistic example value:

```suggestion
    <script>
    let userName = '{{ user.name|escapejs }}'
    ...
```

Maybe we could have a broken example too like `let userDisplay = 'User {{ user.display_name|escapejs }}'`

Reply:
The example improvement is perfectly reasonable, of course, but the "don't do this" example just opens a whole can of worms, because in fact that usage is perfectly safe, but we're not mentioning that because the rules as to what is and isn't safe are too complex to explain so it's better to just say "only do \<this>", not "you can do anything so long as conditions X, Y, Z are met" etc.